solver/llbsolver/bridge.go

```diff
@@ -98,58 +121,93 @@ func (b *llbBridge) loadResult(ctx context.Context, def *pb.Definition, cacheImp
 			return nil, err
 		}
 	}
-	var cms []solver.CacheManager
-	for _, im := range cacheImports {
-		cmID, err := cmKey(im)
+
+	inlineCacheRemotes := map[digest.Digest]*solver.Remote{}
+
+	// Earthly custom inline cache handling. Other cache import types are ignored.
+	for _, cacheImport := range cacheImports {
+		if cacheImport.Type != "registry" {
+			continue
+		}
+
+		if b.importProcessed(cacheImport) {
+			continue
+		}
+
+		err := inBuilderContext(ctx, b.builder, "earthly inline cache", "", func(ctx context.Context, g session.Group) error {
+			inlineCacheRemotes, err = registry.EarthlyInlineCacheRemotes(ctx, b.sm, w, b.registryHosts, g, cacheImport.Attrs)
```

Comment:
This is where remote manifests (including cache relation info) are fetched for the inline cache source.